### PR TITLE
Accessibility enhancements

### PR DESCRIPTION
**Pull request by plweil**
addresses #751 

Adds the following accessibility enhancements:

- Table header sort buttons. Can be reached via the keyboard. Dynamically labeled "Sort table by [column] in [ascending/descending] order." Buttons are absolutely positioned to cover the entire `th` box. The pseudo-elements for the "sort arrows" are placed on the `button` element instead of the `th`.

- `aria-sort` attribute to inform user of the current sort order. E.g., `aria-sort="ascending"`.

- `th` elements are given an explicit role of `columnheader` to accommodate `aria-sort`.

- "sort arrows" via pseudo-elements w/chevron backgrounds left intact, except `border` color is darkened to improve color contrast.

- Table search: add `label` element to provide accessible name, which is visually-hidden but accessible to screen readers via new `sr-only` css class. Magnifying glass is darkened for color contrast. `placeholder` attribute is set to null by default (placeholders might look good to some, but they are an inadequate substitute for labels, and can cause problems for a range of users.).

- The bottom footer area is changed as little as possible while increasing the font size ; darkening the color; removing Vuetify css for the `select` element; adding a `label` element; separating the pagination info from the Previous and Next page buttons for improved clarity; changing the latter to button elements; change the default button label from 'Prev' to 'Previous'; add some margins, etc. to clean up appearance. Button focus outlining is restored; outlines are important for AT users and should not be removed.

- no changes were made to 'pages' pagination mode. I can't help but think that a solution with page number buttons instead of manually entering a page number would be better. But I don't have enough time to work on that right now.

- My chief concern here is accessibility and UI clarity. I don't have a strong opinion regarding the darker colors added here; these colors at least provide an example of better contrast. See, for example [the WET project](https://wet-boew.github.io/v4.0-ci/demos/tables/tables-en.html#wb-auto-1).

### CODE REVIEW

**Review by xaksis — COMMENTED** [review]
hey @plweil have you tested these changes on IE? A while back we had a lot of issues with wrapping rows/tds in templates because IE wouldn't render them.
here's the relevant issue https://github.com/xaksis/vue-good-table/issues/138
I would like to drop support for IE in the near future but for now there are still a lot of people using it there.

**Comment by p0psicles** [comment]
@xaksis tbh people shouldn't be using IE anymore. And you could choose to bump to major  version, when dropping IE support. That way projects that choose to keep supporting IE can lock the version. And we can move on.

**Comment by plweil** [comment]
Thanks for taking a look at this.

No, I have not — I don’t normally test for IE any more, and am not particularly well-set up for it. I’ll see whether I can find a relatively easy way to do it.

pw

**Comment by plweil** [comment]
To support IE or not? Your call. If you’d like me to test this pr in IE, I’ll try to do that. -pw

**Comment by xaksis** [comment]
@p0psicles you're right. Dropping IE support is the logical thing to do for the next major version. @plweil I very much want most of these changes in the current build. From what I can tell, the only thing that'd cause problems is the changes where you wrap td contents in custom components/templates. Is it possible to create a PR without that change? We can save that one for when we bump up the major. 
Also, if you could remove the built files from the PR that'd be great - it'll always cause conflict otherwise.

File: src/components/Table.vue [hunk 1]
```diff
@@ -231,14 +239,13 @@
                   :formattedRow="formattedRow(row)"
                   :index="index"
                 >
-                  <span v-if="!column.html">
+                  <template v-if="!column.html">
```
Comment (xaksis) [inline src/components/Table.vue:242]: This bit here is what I'm talking about.

**Comment by plweil** [comment]
@xasis Thanks. I’ll look into this when I have a free moment.

**Comment by plweil** [comment]
@xaksis I'm just getting back to this. Sorry for the delayed response. I'm aware of problems with IE and the <template> element, but I always thought that these occur when <template> is used outside the <template> section of a component. But no matter. Removing the <template> wrappers in the lines you cite should be fine. It'll just wrap those cell contents in an extra <span>, which is not optimal but no big deal. I'll make that change and remove the built files from the PR shortly.

**Comment by plweil** [comment]
@xaksis do you need anything else from me to merge this, or is this ok? I did remove the built files.

**Comment by xaksis** [comment]
hey @plweil sorry, the PR seems to still include built files and is showing me a bunch of conflicts there so wasn't sure if you were done. Do you mind taking a look maybe it wasn't committed?

**Comment by plweil** [comment]
Will do.

**Comment by plweil** [comment]
Sorry if I'm being dense about this, but I thought I deleted the built files in commit [067cabe](https://github.com/xaksis/vue-good-table/pull/752/commits/067cabe62eb9dee9acccc416f68c89f88faf29c4). What am I doing wrong?

**Comment by p0psicles** [comment]
You shouldn't delete them. You should copy the files from master into your branch. That way there is no change to them.

**Comment by plweil** [comment]
Oh jeez. Thank you @p0psicles. I've re-added the built files from master.

**Comment by plweil** [comment]
@xaksis All good now?

**Comment by xaksis** [comment]
@plweil done! Thank you for your contribution and a very Happy New Year! 
A side note, there were some critical style issues being caused by default user-agent styles of various browsers for buttons. Which I normalized so hopefully version 2.21.2 is good 👍

**Comment by plweil** [comment]
@xaksis good deal and thank you! Happy New Year to you too!

**Comment by plweil** [comment]
@xaksis I'm working on a followup a11y PR -- do I still need to account for IE support?

**Pull request merged**